### PR TITLE
Implement UserProfilesResource, VisitorsResource and ContactsResource

### PR DESCRIPTION
**Pull request by dhartwich1991**
Implement UserProfilesResource, VisitorsResource and ContactsResource

To review: 
- [x] @serj-lotutovici 

List of Commits in this: 

commit 0431750538dada50ec1c30cc3ada6c40908ea250
Author: Daniel Hartwich daniel_hartwich@icloud.com
Date:   Thu Dec 10 17:48:12 2015 +0100

```
Implement ContactsResource and ProfileVisitsResource
```

commit b3c9274961bcef32d35675deeb1e9eedff63a4c1
Author: Daniel Hartwich daniel_hartwich@icloud.com
Date:   Thu Dec 10 09:58:48 2015 +0100

```
Fix some checkstyle warnings
```

commit a735428ad1d972996db4cb1c4c5768bfd3950a6a
Author: Daniel Hartwich daniel_hartwich@icloud.com
Date:   Thu Dec 10 09:54:17 2015 +0100

```
add a lot of javadoc
```

### CODE REVIEW

File: sample/src/main/java/com/xing/android/sdk/sample/ProfileActivity.java [hunk 1]
```diff
@@ -245,9 +247,22 @@ protected XingUser doInBackground(Activity... params) {
...
-
+            ProfileEditingResource editingResource = api.resource(ProfileEditingResource.class);
             Response<XingUser, Object> response = null;
             try {
+                editingResource.updateUserGeneralInformation()
+                      .formField("academic_title", "Ing.").enqueue(new Callback<String, String>() {
+                    @Override
+                    public void onResponse(Response<String, String> response) {
+                        if (response.isSuccess())
+                        Log.i("Update res", response.raw().toString());
```
Comment (serj-lotutovici) [inline sample/src/main/java/com/xing/android/sdk/sample/ProfileActivity.java]: Should be ether in the same line with the if... of surrounded by brackets.

File: sdk/src/main/java/com/xing/android/sdk/model/ContactPaths.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import com.squareup.moshi.Json;
+import com.xing.android.sdk.model.user.XingUser;
+
+import java.util.List;
+
+/**
+ * @author daniel.hartwich
+ */
+public class ContactPaths {
```
Comment (serj-lotutovici) [inline sdk/src/main/java/com/xing/android/sdk/model/ContactPaths.java:33]: Note to self. This object requires a custom json adapter.

File: sdk/src/main/java/com/xing/android/sdk/resources/ContactsResource.java [hunk 3]
```diff
@@ -22,17 +22,380 @@
...
+     * requests within 5 days.
+     * <p></p>
+     * If you try to revoke more contact requests this call will reply with ACCESS_DENIED and you have to wait for 7
+     * days until you can start revoking requests again.
+     * <p></p>
+     *
+     * @param recipientId Recipient ID
+     * @param senderId Sender ID
+     */
+    public CallSpec<String, String> denyOrRevokeContactRequesdt(String recipientId, String senderId) {
```
Comment (serj-lotutovici) [inline sdk/src/main/java/com/xing/android/sdk/resources/ContactsResource.java]: Reques**d**t?

File: sdk/src/main/java/com/xing/android/sdk/resources/ContactsResource.java [hunk 4]
```diff
@@ -22,17 +22,380 @@
...
+     * <td><b>user_fields</b></td>
+     * <td>Comma-separated list of user attributes to return in nested user objects (along with the ID). If empty,
+     * only the ID will be returned.</td>
+     * </tr>
+     * </table>
+     *
+     * @param userId ID of the user whose contact path(s) are to be returned
+     * @param otherUserId ID of any other XING user
+     */
+    public CallSpec<ContactPaths, Object> getContactPath(String userId, String otherUserId) {
```
Comment (serj-lotutovici) [inline sdk/src/main/java/com/xing/android/sdk/resources/ContactsResource.java]: may be getter to say `getContactPaths`?

File: sdk/src/main/java/com/xing/android/sdk/task/contact_petition/IncomingContactPetitionsTask.java [hunk 5]
```diff
@@ -40,7 +40,7 @@
  * @see ContactPetitionRequests#incomingContactPetitions(String, Integer, Integer, List)
  */
 @SuppressWarnings("unused")
-public class IncomingContactPetitionsTask extends PaginatedWithOffsetTask<List<ContactPetition>> {
+public class IncomingContactPetitionsTask extends PaginatedWithOffsetTask<List<ContactRequest>> {
```
Comment (serj-lotutovici) [inline sdk/src/main/java/com/xing/android/sdk/task/contact_petition/IncomingContactPetitionsTask.java:43]: Waiting for the day when we delete this stuff.

**Comment by serj-lotutovici** [comment]
LGTM, aside from a few formatting issues.

File: sdk/src/main/java/com/xing/android/sdk/task/contact_petition/IncomingContactPetitionsTask.java (hunk 5 above)
Comment (dhartwich1991) [inline sdk/src/main/java/com/xing/android/sdk/task/contact_petition/IncomingContactPetitionsTask.java:43]: Me too my friend, me too :+1: :heart_eyes: :shipit:

**Comment by dhartwich1991** [comment]
fixed the above issues

**Pull request merged**